packages/base/src/panelview/objectproperties.tsx

```diff
@@ -256,98 +234,66 @@ class ObjectPropertiesReact extends React.Component<IProps, IStates> {
-      selectedObjectSource: selectedObjectSourceId,
-      schema,
-      selectedObjectSourceData,
-      sourceSchema
-    };
-  }
+    // We selected a source and only show a form for it
+    if (!selectedObjSource) {
+      return (
+        <div>
+          <h3>Layer Properties</h3>
+          <LayerPropertiesForm
+            parentType="panel"
+            model={model}
+            filePath={`${this.state.filePath}::panel`}
+            schema={schema}
+            sourceData={selectedObjectData}
+            syncData={(properties: { [key: string]: any }) => {
+              this.syncObjectProperties(this.state.selectedObject, properties);
+            }}
+            syncSelectedField={this.syncSelectedField}
+          />
+        </div>
+      );
+    }
 
-  render(): React.ReactNode {
-    return this.state.schema && this.state.selectedObjectData ? (
+    // We selected a layer, we show a form for the layer and one for its source
+    const sourceSchema = this._formSchema.get(selectedObjSource.type);
+    const selectedObjectSourceData = selectedObjSource.parameters;
+
+    return (
       <div>
         <h3>Layer Properties</h3>
-        <ObjectPropertiesForm
+        <LayerPropertiesForm
           parentType="panel"
+          model={model}
           filePath={`${this.state.filePath}::panel`}
-          schema={this.state.schema}
-          sourceData={this.state.selectedObjectData}
+          schema={schema}
+          sourceData={selectedObjectData}
           syncData={(properties: { [key: string]: any }) => {
-            if (properties.source) {
-              const sources = this.props.cpModel.jGISModel?.getSources();
-              if (!sources) {
-                throw Error('Unreachable');
-              }
-
-              for (const source of Object.keys(sources)) {
-                if (sources[source].name === properties.source) {
-                  properties.source = source;
-                  break;
-                }
-              }
-            }
-
             this.syncObjectProperties(this.state.selectedObject, properties);
           }}
           syncSelectedField={this.syncSelectedField}
         />
-        {this.state.selectedObjectSourceData && this.state.sourceSchema && (
-          <>
-            <h3>Source Properties</h3>
-            <ObjectPropertiesForm
-              parentType="panel"
-              filePath={`${this.state.filePath}::panel`}
-              schema={this.state.sourceSchema}
-              sourceData={this.state.selectedObjectSourceData}
-              syncData={(properties: { [key: string]: any }) => {
-                this.syncObjectProperties(
-                  this.state.selectedObjectSource,
-                  properties
-                );
-              }}
-              syncSelectedField={this.syncSelectedField}
-            />
-          </>
-        )}
+        <h3>Source Properties</h3>
+        <RasterSourcePropertiesForm
+          parentType="panel"
```

Comment:
Should we add a condition on source type now, or when we add support to other source types ?
```ts
{selectedObjSource.type === 'RasterSource' && (
  <RasterSourcePropertiesForm
  ...
 />
)}
```